packages/widget/src/stores/form/createFormStore.ts

```diff
@@ -0,0 +1,227 @@
+    (set, get) => ({
+      defaultValues: valuesToFormValues(formDefaultValues),
+      userValues: valuesToFormValues(formDefaultValues),
+      touchedFields: {},
+      setDefaultValues: (defaultValue) => {
+        const defaultFormValues = valuesToFormValues(defaultValue);
+        set((state) => ({
+          defaultValues: defaultFormValues,
+          userValues: mergeDefaultFormValues(
+            state.userValues,
+            defaultFormValues,
+          ),
+        }));
+      },
+      isTouched: (fieldName: FormFieldNames) =>
+        !!get().userValues[fieldName]?.isTouched,
+      setAsTouched: (fieldName: FormFieldNames) => {
+        const userValues = {
+          ...get().userValues,
+          [fieldName]: {
+            ...get().userValues[fieldName],
+            isTouched: true,
+          },
+        };
+
+        const touchedFields = getUpdatedTouchedFields(userValues);
+
+        set(() => ({
+          userValues,
+          touchedFields,
+        }));
+      },
+      resetField: (fieldName, { defaultValue } = {}) => {
+        if (defaultValue) {
+          const fieldValues = {
+            ...get().defaultValues[fieldName],
+            value: defaultValue,
+          };
+          const defaultValues = {
+            ...get().defaultValues,
+            [fieldName]: { ...fieldValues },
+          };
+          const userValues = {
+            ...get().userValues,
+            [fieldName]: { ...fieldValues },
+          };
+          const touchedFields = getUpdatedTouchedFields(userValues);
+
+          set(() => {
+            return {
+              defaultValues,
+              userValues,
+              touchedFields,
+            };
+          });
+        } else {
+          const userValues = {
+            ...get().userValues,
+            [fieldName]: { ...get().defaultValues[fieldName] },
+          };
+          const touchedFields = getUpdatedTouchedFields(userValues);
+
+          set(() => ({
+            userValues,
+            touchedFields,
+          }));
+        }
+      },
+      setFieldValue: (fieldName, value, { isDirty, isTouched } = {}) => {
+        const userValues = {
+          ...get().userValues,
+          [fieldName]: {
+            value,
+            isDirty:
+              isDirty === undefined
+                ? get().userValues[fieldName]?.isDirty
+                : isDirty,
+            isTouched:
+              isTouched === undefined
+                ? get().userValues[fieldName]?.isTouched
+                : isTouched,
+          },
+        };
+
+        const touchedFields = getUpdatedTouchedFields(userValues);
+
+        set(() => ({
+          userValues,
+          touchedFields,
+        }));
+      },
+      getFieldValues: (...names) =>
+        names.map((name) => get().userValues[name]?.value),
+
+      // The following a is a "placeholder" "for validation for toAddress
+      // is will be re-addressed in the next piece of work for Wallet Switching
```

Comment:
Can we please improve this comment? 😄

Reply:
Actually we can probably live without it - I'll delete it